From d3b0cf855ff0b9d0405fa54b11496f228acfd7d3 Mon Sep 17 00:00:00 2001
From: Saahil Tomar <quic_saahtoma@quicinc.com>
Date: Tue, 27 Feb 2024 16:53:57 +0530
Subject: [PATCH] crypto: qce - Resolve [-Werror=enum-conversion] warning.

This change resolves the warnings caused by dma_transfer_direction
and dma_data_direction enums being used interchangeably leading to
warnings due to incompatible type conversions.

Change-Id: I081a2f357e786e27bbfca2ea3c69635c5a3021f1
Signed-off-by: Saahil Tomar <quic_saahtoma@quicinc.com>
---
 drivers/crypto/qce/aead.c     | 4 ++--
 drivers/crypto/qce/dma.c      | 8 ++++++--
 drivers/crypto/qce/sha.c      | 4 ++--
 drivers/crypto/qce/skcipher.c | 4 ++--
 4 files changed, 12 insertions(+), 8 deletions(-)

diff --git a/drivers/crypto/qce/aead.c b/drivers/crypto/qce/aead.c
index 8f938660f5f3..ca47859acab9 100644
--- a/drivers/crypto/qce/aead.c
+++ b/drivers/crypto/qce/aead.c
@@ -52,12 +52,12 @@ static void qce_aead_done(void *data)
 			dma_unmap_sg(qce->dev,
 				qce_bam_txn->qce_reg_read_sgl,
 				qce_bam_txn->qce_read_sgl_cnt,
-				DMA_DEV_TO_MEM);
+				dir_dst);
 		if (qce_bam_txn->qce_write_sgl_cnt)
 			dma_unmap_sg(qce->dev,
 				qce_bam_txn->qce_reg_write_sgl,
 				qce_bam_txn->qce_write_sgl_cnt,
-				DMA_MEM_TO_DEV);
+				dir_src);
 	}
 
 	if (IS_CCM(rctx->flags)) {
diff --git a/drivers/crypto/qce/dma.c b/drivers/crypto/qce/dma.c
index 33d9e0eed9f4..92014e1dc9fb 100644
--- a/drivers/crypto/qce/dma.c
+++ b/drivers/crypto/qce/dma.c
@@ -32,15 +32,19 @@ static int qce_dma_prep_cmd_sg(struct qce_device *qce, struct dma_chan *chan,
 {
 	struct dma_async_tx_descriptor *dma_desc;
 	dma_cookie_t cookie;
+	enum dma_data_direction data_dir;
 	struct qce_desc_info *desc;
 
+	data_dir = (dir == DMA_MEM_TO_DEV) ? DMA_TO_DEVICE :
+		(dir == DMA_DEV_TO_MEM) ? DMA_FROM_DEVICE : DMA_BIDIRECTIONAL;
+
 	desc = qce->dma.qce_bam_txn->qce_desc;
 
 	if (!qce_bam_sgl || !qce_sgl_cnt)
 		return -EINVAL;
 
 	if (!dma_map_sg(qce->dev, qce_bam_sgl,
-				qce_sgl_cnt, dir)) {
+				qce_sgl_cnt, data_dir)) {
 		dev_err(qce->dev, "failure in mapping sgl for cmd desc\n");
 		return -ENOMEM;
 	}
@@ -49,7 +53,7 @@ static int qce_dma_prep_cmd_sg(struct qce_device *qce, struct dma_chan *chan,
 						dir, flags);
 	if (!dma_desc) {
 		pr_err("%s:failure in prep cmd desc\n",__func__);
-		dma_unmap_sg(qce->dev, qce_bam_sgl, qce_sgl_cnt, dir);
+		dma_unmap_sg(qce->dev, qce_bam_sgl, qce_sgl_cnt, data_dir);
 		kfree(desc);
 		return -EINVAL;
 	}
diff --git a/drivers/crypto/qce/sha.c b/drivers/crypto/qce/sha.c
index 84459385feef..9d892a68fd91 100644
--- a/drivers/crypto/qce/sha.c
+++ b/drivers/crypto/qce/sha.c
@@ -62,12 +62,12 @@ static void qce_ahash_done(void *data)
 			dma_unmap_sg(qce->dev,
 				qce_bam_txn->qce_reg_read_sgl,
 				qce_bam_txn->qce_read_sgl_cnt,
-				DMA_DEV_TO_MEM);
+				DMA_FROM_DEVICE);
 		if (qce_bam_txn->qce_write_sgl_cnt)
 			dma_unmap_sg(qce->dev,
 				qce_bam_txn->qce_reg_write_sgl,
 				qce_bam_txn->qce_write_sgl_cnt,
-				DMA_MEM_TO_DEV);
+				DMA_TO_DEVICE);
 	}
 
 	req->src = rctx->src_orig;
diff --git a/drivers/crypto/qce/skcipher.c b/drivers/crypto/qce/skcipher.c
index ef2f86393bf9..7aa8eee163bb 100644
--- a/drivers/crypto/qce/skcipher.c
+++ b/drivers/crypto/qce/skcipher.c
@@ -62,12 +62,12 @@ static void qce_skcipher_done(void *data)
 			dma_unmap_sg(qce->dev,
 				qce_bam_txn->qce_reg_read_sgl,
 				qce_bam_txn->qce_read_sgl_cnt,
-				DMA_DEV_TO_MEM);
+				dir_dst);
 		if (qce_bam_txn->qce_write_sgl_cnt)
 			dma_unmap_sg(qce->dev,
 				qce_bam_txn->qce_reg_write_sgl,
 				qce_bam_txn->qce_write_sgl_cnt,
-				DMA_MEM_TO_DEV);
+				dir_src);
 	}
 
 	memcpy(rctx->iv, result_buf->encr_cntr_iv, rctx->ivsize);
-- 
2.34.1

